text/trace/0107-sampling-priority.md

```diff
@@ -0,0 +1,81 @@
+# Associating sampling priority with the trace using tracestate
+
+Propagating `sampling.priority` (calculated by probability sampler) in a
+`tracestate` will enable a better experience for system with the independent
+stateless configuration of components sampling.
+
+## Motivation
+
+Consistent sampling decision made in each app of a distributed trace is
+important for better user experience of trace analysis. Consistency is achieved
+by following means:
+- Same hashing algorithms used across all apps in a trace.
+- Sampling flag propagated from the head component/app is used by downstream apps to sample in a given trace.
+
+Increasingly more apps participate in distributed tracing. With the 
+standardized wire format for context propagation, there is bigger chance that 
+algorithm for sampling chosen by one app may not match another app. 
+Since coordination of sampling algorithms across many apps not always possible, 
+OpenTelemetry must provide a way to share sampling hints between apps.
+
+Beyond sampling hints infrastructure, it is beneficial for the community to
+agree on standard behavior and hints exposed by probability sampler.
+
+The `sampling.priority` property is used by many vendors for various purposes.
+Propagating this field alongside the trace will allow for many improvements and
+as a very minimum will simplify transition of customers from SDKs using
+different `trace-id` based hash functions to OpenTelemetry SDK.
+
+Also see related discussions on probability sampler here: <https://github.com/open-telemetry/opentelemetry-specification/pull/570>
+
+## Explanation
+
+### Sampling hints exchange
+
+OpenTelemetry SDK has an infrastructure to expose sampling hints as span attributes today
+via
+[SamplingResult](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shouldsample)
+class.
+
+This recommendation suggests OpenTelemetry will allow to return an updated
+`tracestate` that will be used for a newly created `Span`. This will allow
```

Comment:
I think this is mixing two different concerns, the API of the sampler result and the serialization format of the context. Not everybody is going to use W3C TraceContext for propagation format, so the mechanism of returning the probability should be somehow separate from the propagation format (i.e. the sampling probability may be recorded in the Span or SpanContext).

Reply:
Agreed. We can make changes for this.